tests/components/fritz/test_config_flow.py

```diff
@@ -0,0 +1,408 @@
+        side_effect=fc_class_mock,
+    ), patch("homeassistant.components.fritz.common.FritzStatus"):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_SSDP}, data=MOCK_SSDP_DATA
+        )
+        assert result["type"] == RESULT_TYPE_ABORT
+        assert result["reason"] == "already_configured"
+
+
+async def test_ssdp_already_configured_host(hass: HomeAssistant, fc_class_mock):
+    """Test starting a flow from discovery but no device found."""
+
+    mock_config = MockConfigEntry(
+        domain=DOMAIN,
+        data=MOCK_USER_DATA,
+        unique_id="different-test",
+    )
+    mock_config.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.fritz.common.FritzConnection",
+        side_effect=fc_class_mock,
+    ), patch("homeassistant.components.fritz.common.FritzStatus"):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_SSDP}, data=MOCK_SSDP_DATA
+        )
+        assert result["type"] == RESULT_TYPE_ABORT
+        assert result["reason"] == "already_configured"
+
+
+async def test_ssdp_already_configured_host_uuid(hass: HomeAssistant, fc_class_mock):
+    """Test starting a flow from discovery but no device found."""
+
+    mock_config = MockConfigEntry(
+        domain=DOMAIN,
+        data=MOCK_USER_DATA,
+        unique_id=None,
+    )
+    mock_config.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.fritz.common.FritzConnection",
+        side_effect=fc_class_mock,
+    ), patch("homeassistant.components.fritz.common.FritzStatus"):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_SSDP}, data=MOCK_SSDP_DATA
+        )
+        assert result["type"] == RESULT_TYPE_ABORT
+        assert result["reason"] == "already_configured"
+
+
+async def test_ssdp_already_in_progress_host(hass: HomeAssistant, fc_class_mock):
+    """Test starting a flow from discovery twice."""
+    with patch(
+        "homeassistant.components.fritz.common.FritzConnection",
+        side_effect=fc_class_mock,
+    ), patch("homeassistant.components.fritz.common.FritzStatus"):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_SSDP}, data=MOCK_SSDP_DATA
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "confirm"
+
+        MOCK_NO_UNIQUE_ID = MOCK_SSDP_DATA.copy()
+        del MOCK_NO_UNIQUE_ID[ATTR_UPNP_UDN]
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_SSDP}, data=MOCK_NO_UNIQUE_ID
+        )
+        assert result["type"] == RESULT_TYPE_ABORT
+        assert result["reason"] == "already_in_progress"
+
+
+async def test_ssdp(hass: HomeAssistant, fc_class_mock):
+    """Test starting a flow from discovery but no device found."""
+    with patch(
```

Comment:
patch `async_setup_entry` here and verify it gets called

Reply:
Done